src/lib.rs

```diff
@@ -89,6 +97,23 @@ impl Service<Request<Body>> for MetricProvider {
     }
 }
 
+fn is_auth_token_valid(secret: &str, headers: &HeaderMap<HeaderValue>) -> bool {
+    match headers.get(AUTHORIZATION) {
+        Some(value) => value
+            .to_str()
+            .ok()
+            .map(|t| {
+                if let Some(t) = t.strip_prefix("Bearer") {
+                    t.trim().eq(secret)
+                } else {
+                    false
+                }
+            })
+            .unwrap(),
```

Comment:
This will crash the server if someone sends a non-utf8 value for the `Authorization` header.

Reply:
so i'm thinking that refactoring by removing the call to `.ok()` and change `.map()` to a `.map_or_else()` would be one solution. what you think?

```rust
fn is_auth_token_valid(secret: &str, headers: &HeaderMap<HeaderValue>) -> bool {
    match headers.get(AUTHORIZATION) {
        Some(value) => value
            .to_str()
            .map_or_else(|_| false, |t| {
                if let Some(t) = t.strip_prefix("Bearer ") {
                    t.eq(secret)
                } else {
                    false
                }
            })
            .unwrap(),
        None => false,
    }
}
`

btw.. wanna log the error or theres no need for it?

Reply:
> btw.. wanna log the error or theres no need for it?

I don't think there is a need for it

> I'm also stripping the suffix whitespace from the "Bearer ", and removed the trim() call before comparing the token.

:+1: